platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/pax/web/jetty/CsrfFilter.java

```diff
@@ -120,30 +153,42 @@ public void init(FilterConfig filterConfig) throws ServletException {
   @Override
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
       throws IOException, ServletException {
-    LOGGER.debug("Performing doFilter() on CsrfFilter");
-    HttpServletRequest httpRequest = (HttpServletRequest) request;
-    HttpServletResponse httpResponse = (HttpServletResponse) response;
-    String targetContextPath = httpRequest.getRequestURI();
-    String requestMethod = httpRequest.getMethod();
-    String userAgentHeader = httpRequest.getHeader(HttpHeaders.USER_AGENT);
-
-    // Begin CSRF checks if request is accessing a Cross-Site protected context
-    if (protectedContexts.stream().anyMatch(targetContextPath::startsWith)
-        && doBrowserProtectionFilter(httpRequest, httpResponse, targetContextPath, requestMethod)) {
-      return;
-    }
 
-    // Execute CSRF check if user is accessing /services
-    if (targetContextPath.startsWith(SERVICE_CONTEXT)
-        && doSystemProtectionFilter(
-            httpRequest, httpResponse, targetContextPath, requestMethod, userAgentHeader)) {
-      return;
+    if (shouldFilter) {
+      LOGGER.debug("Performing doFilter() on CsrfFilter");
+      HttpServletRequest httpRequest = (HttpServletRequest) request;
+      HttpServletResponse httpResponse = (HttpServletResponse) response;
+      String targetContextPath = httpRequest.getRequestURI();
+      String requestMethod = httpRequest.getMethod();
+      String userAgentHeader = httpRequest.getHeader(HttpHeaders.USER_AGENT);
+
+      // Begin CSRF checks if request is accessing a Cross-Site protected context
+      if (protectedContexts.stream().anyMatch(targetContextPath::startsWith)
```

Comment:
are there unit tests for this new functionality?

Reply:
I have to add them still